### PR TITLE
Move to repo toolset

### PR DESCRIPTION
**Pull request by jmarolf**
move this repository to build and sign using repotoolset

### CODE REVIEW

**Comment by jmarolf** [comment]
@pjanotti Iooks like the appveyor build is attempting to sign the output
```
Build FAILED.
ALINK : error AL1028: Key file 'c:\Users\appveyor\.nuget\packages\roslyntools.repotoolset\1.0.0-beta-62522-01\tools\35MSSharedLib1024.snk' is missing the private key needed for signing [C:\projects\platform-compat\src\Microsoft.DotNet.Analyzers.Compatibility\Microsoft.DotNet.Analyzers.Compatibility.csproj]
    0 Warning(s)
    1 Error(s)
Time Elapsed 00:01:28.21
Command exited with code 1
```

The plan would be for signed builds to come from [here](https://devdiv.visualstudio.com/DevDiv/_build/index?context=allDefinitions&path=\&definitionId=8440).  Do you have permissions to modify the appveyor build or can you give me permissions?

**Comment by pjanotti** [comment]
@jmarolf Unfortunately I don't have the rights @terrajobst is the guy that owns that appveyor project.

**Comment by jmarolf** [comment]
@pjanotti alright thanks.  @terrajobst is on vacation still (I think) but I'll ping him.

**Comment by terrajobst** [comment]
Just downloaded your PR and tried to build it locally. I get this:

```
$ .\build.cmd
Downloading vswhere
Cannot bind argument to parameter 'Path' because it is null.
System.Management.Automation.ParameterBindingValidationException: Cannot bind argument to parameter 'Path' because it is null.
   at System.Management.Automation.ExceptionHandlingOps.CheckActionPreference(FunctionContext funcContext, Exception exception)
   at System.Management.Automation.Interpreter.ActionCallInstruction`2.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
   at System.Management.Automation.Interpreter.EnterTryCatchFinallyInstruction.Run(InterpretedFrame frame)
at LocateVisualStudio, P:\platform-compat\build\Build.ps1: line 90
at <ScriptBlock>, P:\platform-compat\build\Build.ps1: line 154
at <ScriptBlock>, <No file>: line 1
```

Do I have sacrifice a checking first?

**Comment by jmarolf** [comment]
@terrajobst I've gotten everything working.  The plan would be to check this in as-is so appveyor keeps working and then I submit a subsequent PR that moves this repo to the dotnet infrastructure.

File: appveyor.yml [hunk 1]
```diff
@@ -0,0 +1,41 @@
+image: Visual Studio 2017
```
[the remark this answers is not in the document]
Comment (jmarolf) [reply appveyor.yml]: This is throwaway work to make appveyor pass  I'll delete this file once I move everything to Jenkins

File: Directory.Build.props [hunk 2]
```diff
@@ -0,0 +1,17 @@
+<!-- Copyright (c)  Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information. -->
+<Project>
+  <Import Project="build\NuGet.props"/>
+  <Import Project="build\Versions.props"/>
```
[the remark this answers is not in the document]
Comment (jmarolf) [reply Directory.Build.props]: `eng` I believe is the new syntax, I'll change it.

[the remark this answers is not in the document]
Comment (jmarolf) [reply Directory.Build.props]: Actually I can't find a repo that uses this convention.  I'll change it when other repos start migrating.

**Comment by terrajobst** [comment]
Thanks a ton @jmarolf!

**Pull request merged**